src/ngraph/env_tbl.hpp

```diff
@@ -0,0 +1,197 @@
+//*****************************************************************************
+// Copyright 2017-2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License")
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+// This collection contains one entry for each environment variable.If an environment variable is
+// added it must be added to this list.
+//
+// In order to use this list you want to define a macro named exactly NGRAPH_DEFINE_ENVVAR
+// When you are done you should undef the macro
+
+#ifndef NGRAPH_DEFINE_ENVVAR
+#warning "NGRAPH_DEFINE_ENVVAR not defined"
+#define NGRAPH_DEFINE_ENVVAR(ENUMID, NAME, DEFAULT, DESCRIPTION)
+#endif
+
+NGRAPH_DEFINE_ENVVAR(ngraph::EnvVarEnum::NGRAPH_CODEGEN,
```

Comment:
Call these `DEFINE_ENVAR` since `ngraph` comes from the namespace, and defining things should start with `def` not `ngr`. Also, consider `DEFINE_BOOL_ENVAR` etc. so you don't need strings like "FALSE".